torch/_meta_registrations.py

```diff
@@ -124,6 +126,33 @@ def meta_copy_(self, src, non_blocking=False):
     return self
 
 
+@register_meta(aten.resize_.default)
+def meta_resize_(self, size, *, memory_format=None):
+    if same_shape(self.shape, size):
+        return self  # ignores memory_format
```

Comment:
This implements behavior from the docs: "Note that memory format of self is going to be unaffected if self.size() matches sizes."

However, ATen always uses `memory_format`. `_resize_impl_` early exits, but `memory_format` is used later anyway:

```cpp
template <typename T>
const Tensor& _resize_(
    const Tensor& self,
    ArrayRef<T> size,
    c10::optional<MemoryFormat> optional_memory_format) {
  auto* self_ = self.unsafeGetTensorImpl();
  // NOLINTNEXTLINE(bugprone-argument-comment)
  _resize_impl_<T>(self_, size, /*strides=*/c10::nullopt, true);
  if (optional_memory_format.has_value()) {
    auto memory_format =
        optional_memory_format.value();
    TORCH_CHECK(
        memory_format != MemoryFormat::Preserve,
        "Unsupported memory format",
        memory_format);
    self_->empty_tensor_restride(memory_format);
  }
  return self;
}
```

```
[ins] In [6]: torch._meta_registrations.meta_resize_(torch.randn(3, 4, 5, 6), (3, 4, 5, 6), memory_format=torch.channels_last).is_contiguous()
Out[6]: True

[ins] In [7]: torch.Tensor.resize_(torch.randn(3, 4, 5, 6), (3, 4, 5, 6), memory_format=torch.channels_last).is_contiguous()
Out[7]: False

[ins] In [8]: torch.Tensor.resize_(torch.randn(3, 4, 5, 6), (3, 4, 5, 6), memory_format=torch.channels_last).is_contiguous(memory_format=torch.channels_last)
Out[8]: True

```